employees.rb

```diff
@@ -0,0 +1,159 @@
+class Employee
+  attr_accessor:full_name
+  attr_accessor:id
+
+  def initialize(full_name, id)
+    @full_name = full_name
+    @id = id
+  end
+
+  def surname
+    @full_name.split(' ', 2).last
+  end
+end
+
+class Programmer < Employee
+  attr_accessor:languages
+
+  def initialize(full_name, id, languages)
+    super(full_name, id)
+    @languages = languages
+  end
+end
+
+class OfficeManager < Employee
+  attr_accessor:office
+
+  def initialize(full_name, id, office)
+    super(full_name, id)
+    @office = office
+  end
+end
+
+
+
+
+def add_employee(employees)
+  puts '[Add an employee]'
+  print 'Full name: '
+  full_name = gets.chomp
+  print 'ID: '
+  id = gets.chomp
+  print "Is the person [e]mployee, [p]rogrammer or an [o]ffice menager? "
+  role = gets.chomp.downcase
+  case role
+  when role='e' then
+    employees<<Employee.new(full_name, id)
+  when role='o' then
+    print "Office: "
+    employees<<OfficeManager.new(full_name, id, gets.chomp)
+  when role='p'
+    print "Programming languages: "
+    employees<<Programmer.new(full_name, id, gets.chomp)
+  else
+    puts "Wrong input"
+    return
+  end
+end
+
+def view_employees(employees)
+  print "Sort by [f]isrt name or [l]ast name: "
+  sort=gets.chomp
+  
+  case sort
```

Comment:
The `l` and `f` branch are nearly identical is there a way to remove this code duplication?